nixos/modules/services/games/teeworlds.nix

```diff
@@ -1,96 +1,116 @@
-        type = types.nullOr types.str;
-        default = null;
-        description = lib.mdDoc ''
-          Password to connect to the server.
-        '';
-      };
-
-      rconPassword = mkOption {
-        type = types.nullOr types.str;
-        default = null;
-        description = lib.mdDoc ''
-          Password to access the remote console. If not set, a randomly generated one is displayed in the server log.
-        '';
-      };
-
-      port = mkOption {
-        type = types.int;
-        default = 8303;
-        description = lib.mdDoc ''
-          Port the server will listen on.
-        '';
-      };
+  imports = [
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "name" ]
+      [ "services" "teeworlds" "settings" "sv_name" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "register" ]
+      [ "services" "teeworlds" "settings" "sv_register" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "motd" ]
+      [ "services" "teeworlds" "settings" "sv_motd" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "password" ]
+      [ "services" "teeworlds" "settings" "password" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "rcon_password" ]
+      [ "services" "teeworlds" "settings" "sv_rcon_password" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "port" ]
+      [ "services" "teeworlds" "settings" "sv_port" ])
+    (mkRemovedOptionModule [ "services" "teeworlds" "extraOptions" ]
+      "Use services.teeworlds.settings instead.")
+  ];
+
+  options.services.teeworlds = {
+    enable = mkEnableOption (lib.mdDoc "Teeworlds Server");
+
+    openPorts = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to open firewall ports for Teeworlds
+      '';
+    };
 
-      extraOptions = mkOption {
-        type = types.listOf types.str;
-        default = [];
-        description = lib.mdDoc ''
-          Extra configuration lines for the {file}`teeworlds.cfg`. See [Teeworlds Documentation](https://www.teeworlds.com/?page=docs&wiki=server_settings).
-        '';
-        example = [ "sv_map dm1" "sv_gametype dm" ];
+    settings = mkOption {
+      default = {};
+      example = literalExpression ''
+        {
+          sv_motd = "Welcome to this teeworlds ctf server!";
+          sv_map = "ctf1";
+          sv_gametype = "ctf";
+        }
+      '';
+      description = lib.mdDoc ''
+        See [Teeworlds Documentation](https://www.teeworlds.com/?page=docs&wiki=server_settings) for all configuration options.
+      '';
+      type = lib.types.submodule {
+        options = {
+          sv_name = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = lib.mdDoc ''
+              Name of the server. Defaults to 'unnamed server'.
+            '';
+          };
+
+          sv_register = mkOption {
+            type = types.int;
```

Comment:
If we change the type, we should use `mkChangedOptionModule`; however here we should keep this as `bool` and map to 0/1 when writing the config.